### ISSUE TITLE
First test on Garden scene

### ISSUE THREAD

**Issue by rebecca-lay3rs** [Lay3rss/MERF#1]
### Editing train.sh

#!/bin/bash

SCENE=garden
DATA_DIR=/root/google-research/merf/nerf_real_360
CHECKPOINT_DIR=~/checkpoints/merf/"$SCENE"

export CUDA_VISIBLE_DEVICES=0
python train.py \
  --gin_configs=configs/merf.gin \
  --gin_bindings="Config.data_dir = '${DATA_DIR}/${SCENE}'" \
  --gin_bindings="Config.checkpoint_dir = '${CHECKPOINT_DIR}'"

**Comment by rebecca-lay3rs** [comment]
### Problem with jax array
**Error**: `arrayimpl has no attribute split`

Had to change the function to split the array (https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.split.html#jax.numpy.split) in http://51.159.152.49:8089/edit/google-research/merf/internal/grid_utils.py L175:

`sparse_grid_y, y = jnp.split(y, [8], axis=1) # y.split([8], axis=1) ######################`

**Comment by rebecca-lay3rs** [comment]
### Problem loading cameras and images
- input_path problem, it is always None in original code
- First test: I hardcode the input path in the code

```
    self.load_cameras(input_file="/root/google-research/merf/nerf_real_360/garden/sparse/0/")
    self.load_images(input_file="/root/google-research/merf/nerf_real_360/garden/sparse/0/")
```

- and also here (merf/internal/pycolmap/pycolmap/scene_manager.py L29): 

`        self.image_path = "/root/google-research/merf/nerf_real_360/garden/images" # None`

- changed following functions in http://51.159.152.49:8089/edit/google-research/merf/internal/pycolmap/pycolmap/scene_manager.py :
```

    def load_images(self, input_file=None): ###################################
        if input_file is not None:
            #input_file = self.folder + 'images.bin'
            if os.path.exists(input_file):
                self._load_images_bin(input_file+'images.bin')
            else:
                #input_file = self.folder + 'images.txt'
                if os.path.exists(input_file):
                    self._load_images_txt(input_file+'images.txt')
                else:
                    raise IOError('no images file found')

```
```
    def load_cameras(self, input_file=None): ###################################
        if input_file is not None:
            # print("BLAAAAAAAAAAAA", input_file) ############## /root/google-research/merf/nerf_real_360/garden/sparse/0/
            #input_file = self.folder + 'cameras.bin'
            if os.path.exists(input_file):
                self._load_cameras_bin(input_file + 'cameras.bin')
            else:
                #input_file = self.folder + 'cameras.txt'
                if os.path.exists(input_file):
                    self._load_cameras_txt(input_file + 'cameras.txt')
                else:
                    raise IOError('no cameras file found')
```

```
    ################################### TODO CHANGE THIS FUNCTION??? REBECCA
    def load_points3D(self, input_file=None):
        if input_file is None:
            input_file = self.folder + 'points3D.bin'
            if os.path.exists(input_file):
                self._load_points3D_bin(input_file)
            else:
                input_file = self.folder + 'points3D.txt'
                if os.path.exists(input_file):
                    self._load_points3D_txt(input_file)
                else:
                    raise IOError('no points3D file found')
```

**Comment by rebecca-lay3rs** [comment]
### Webviewer

- added `launch_webserver.sh` and `launch_webserver_utils.sh` to launch http-server from BASH (obtained from SMERF repo) but always get the following error:

```
[2024-03-01T17:39:21.918Z]  "GET /root/google-research/merf/checkpoints/garden/baked/scene_params.json" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
[2024-03-01T17:39:21.919Z]  "GET /root/google-research/merf/checkpoints/garden/baked/scene_params.json" Error (404): "Not found"
[2024-03-01T17:39:21.929Z]  "GET /root/google-research/merf/checkpoints/garden/baked/occupancy_grid_8.png" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
[2024-03-01T17:39:21.929Z]  "GET /root/google-research/merf/checkpoints/garden/baked/occupancy_grid_8.png" Error (404): "Not found"
[2024-03-01T17:39:21.943Z]  "GET /root/google-research/merf/checkpoints/garden/baked/occupancy_grid_16.png" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
[2024-03-01T17:39:21.944Z]  "GET /root/google-research/merf/checkpoints/garden/baked/occupancy_grid_32.png" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
[2024-03-01T17:39:21.944Z]  "GET /root/google-research/merf/checkpoints/garden/baked/occupancy_grid_64.png" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
[2024-03-01T17:39:21.944Z]  "GET /root/google-research/merf/checkpoints/garden/baked/occupancy_grid_128.png" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
[2024-03-01T17:39:21.944Z]  "GET /root/google-research/merf/checkpoints/garden/baked/occupancy_grid_16.png" Error (404): "Not found"
[2024-03-01T17:39:21.945Z]  "GET /root/google-research/merf/checkpoints/garden/baked/occupancy_grid_32.png" Error (404): "Not found"
[2024-03-01T17:39:21.945Z]  "GET /root/google-research/merf/checkpoints/garden/baked/occupancy_grid_64.png" Error (404): "Not found"
[2024-03-01T17:39:21.945Z]  "GET /root/google-research/merf/checkpoints/garden/baked/occupancy_grid_128.png" Error (404): "Not found"
```

- using `python -m http.server --directory ~/checkpoints/merf/garden/baked/` the viewer does not appear, I get only the list of folders
- if no directory is put inside the python command, we have:

dir is a required parameter.

To view a MERF scene, specify the following parameters in the URL:
(Required) The URL to a MERF scene directory.
s: (Optional) The dimensions as width,height. E.g. 640,360.
vfovy:  (Optional) The vertical field of view of the viewer.

**Comment by rebecca-lay3rs** [comment]
#### Cloitre_Gal2_E2_Statues_proche
- did not work, one of the losses was infinite and no video is saved in the end
- the MERF code reads the COLMAP data directly from the cameras.bin and images.bin (or .txt), using the `pycolmap` code that I had to change
